include/gc.h

```diff
@@ -7,49 +7,36 @@
 #include <vector>
 
 namespace yapvm {
-namespace gc {
+namespace ygc {
 
 using namespace yapvm::yobjects;
 
-// wrapper for YObject
-class GCObject {
-    YObject *obj_;
-    bool is_marked_;
-    std::vector<GCObject *> kids_;
-
-public:
-    GCObject(YObject *val);
-
-    bool &is_marked();
-    YObject *obj();
-    std::vector<GCObject *> &kids();
-};
-
 // TODO
-class Scope {
+class YScope {
 public:
-    std::map<std::string, GCObject *> m_;
```

Comment:
use managed pointers instead of raw, if need to share - then shared_ptr is your case, if not - unique_ptr or scoped_ptr